### PR TITLE
Add a Contributing.md file

### PR DESCRIPTION
**Pull request by aloisdg**
Move from #212

### CODE REVIEW

**Comment by stieben** [comment]
What is missing for a merge here?

**Comment by aloisdg** [comment]
@stieben I rewrote [one](https://github.com/alexbooker/base16-builder/blob/master/contributing.md) yesterday for another project. I will push some changes here later. I think I can say that it is 90% ready, at least for now.

If you want you can check my other one and make some modifications. It could be nice to have another point of view on this. :)

**Comment by aloisdg** [comment]
I think we can merge it.

**Comment by stp-ip** [comment]
The PR is not as clean as I would like it to be. Only your changes should be merged in with this PR. The other changes need their own PR.
Still reading through the contribution changes. Quite great.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* Search if the [issue](https://github.com/jsonresume/resume-schema/issues) exist.
+* If the issue exists, then contribute by comments, or you can create one.
+* Try to be as descriptive, concise and specific as possible.
+* Titles must be able to stand on their own without requiring readers to click on the issue.
+* Submit your issue.
+
+## Creating a pull request
+
+* Check if there is an issue related to your feature. If there is not, create one.
+* Search for an [pull request](https://github.com/jsonresume/resume-schema/pulls) that relates to your submission. You don't want to duplicate effort.
```
Comment (stp-ip) [inline CONTRIBUTING.md]: Shouldn't that be "a [pull request]"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+* [Fork the project](https://help.github.com/articles/fork-a-repo/).
+* Create a [pull request](https://help.github.com/articles/creating-a-pull-request/).
+* Submit your pull request.
```
Comment (stp-ip) [inline CONTRIBUTING.md]: * The pull request should be self contained and provide only changes corresponding to the issue/PR title.
(Or something like that)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Tips & Tricks for a good contribution
+
+* Be consistent with resume-schema's lexicon and [schema.org](http://schema.org/)
+* Keep it simple, stupid
+* Don't repeat yourself
+* Think about localization
```
Comment (stp-ip) [inline CONTRIBUTING.md]: * Keep it generic. Json resume thrives to be a generic standard for everyone's resume.
(not sure about the wording)

**Comment by stp-ip** [comment]
Besides the comments, I would say awesome. Thanks for the time you put into it. Will be quite valuable for v1 contributions.

File: CONTRIBUTING.md (hunk 1 above)
Comment (aloisdg) [inline CONTRIBUTING.md]: Where?

File: CONTRIBUTING.md (hunk 3 above)
Comment (aloisdg) [inline CONTRIBUTING.md]: So we let go the by dev for dev, right?

File: CONTRIBUTING.md (hunk 2 above)
Comment (aloisdg) [inline CONTRIBUTING.md]: Added. I have to clean this PR btw.

File: CONTRIBUTING.md (hunk 3 above)
Comment (stp-ip) [inline CONTRIBUTING.md]: I would say so. In most of the discussions we tried to move the standard to be as generic as possible. The tooling can still be by dev for dev.

File: CONTRIBUTING.md (hunk 1 above)
Comment (stp-ip) [inline CONTRIBUTING.md]: Search for a(n) [pull request] <- L20

File: CONTRIBUTING.md (hunk 3 above)
Comment (aloisdg) [inline CONTRIBUTING.md]: Should be written somewhere official.

related to #219

**Comment by olivif** [comment]
agree with @stp-ip, let's take out all the other commits from this PR, keep only contributing.md changes. Do we have a convention to squash commits before merging? Makes things a bit clearer.

**Comment by stp-ip** [comment]
I personally dislike squashing into one commit, if reasoning and discussion is removed. So squash down to the least amount of commits needed to tell the history of the feature/PR. So a typo commit would be squashed into another one, but one used writing/coding solution B instead of A, cause it was problematic/came up in discussions etc. should be kept.

**Comment by olivif** [comment]
that makes sense @stp-ip! *meta* should we add that sentence to the Contributing.md ?

**Comment by thomasdavis** [comment]
We could probably merge this is now and patch it over time.

**Comment by stp-ip** [comment]
So we have 3 +1 for a merge. Only missing thing is removing non related commits from the PR. @aloisdg would be great if you could cherry-pick the right commits and push again.

File: CONTRIBUTING.md (hunk 1 above)
Comment (aloisdg) [inline CONTRIBUTING.md]: Ok got it.

**Comment by aloisdg** [comment]
I rebased the branch.

**Comment by stp-ip** [comment]
Perfect.

**Pull request merged**